docs/deployment/microbit-deployment-draft.md

```diff
@@ -0,0 +1,166 @@
+
+{<< what other things should constitute a breaking change, and should we allow certain kinds of breaking change inside a version of the editor? IE can we support more than one MicroPython version for a particular deployed version of the front-end?>>}}
+
+It may be desirable to agree a high water mark in memory and growing beyond this is considered a breaking change. It would then be necessary to police memory growth but people do already do this I think. This would allow small increases in memory usage to be possible without a breaking change.
+
+### URL versioning scheme
+
+For each published major version, the deployed editor will live at
+
+`python.microbit.org/v<major version>/editor.html`
+
+Because the use of semantic versioning ensures that things will not break for the user when upgrading minor versions, there will not be fine-grained control over which minor version a user gets. 
+
+For example visiting `python.microbit.org/v2/` will serve version `2.<latest minor>`
+
+### Versions in URLs
+
+When a script is published or shared, it must be published including the major version of the editor (and therefore the MicroPython stack) that it was created with so that anyone following the link will be taken back to the version the script was created with.
+
+If a user visits python.microbit.org/latest/ and is working on the editor there, the URL published should still be versioned, such as python.microbit.org/v1/. 
+
+See “workflows” for discussion of how the user will be notified about newer versions
+
+
+### Versioning in hex files
+
+When the user is provided a hex file, the full semantic editor version must also be embedded into the hex file at a known location so that when unpacking the hex, any version of the editor can offer to load the version that the script was created in.
+
+The hex file must also contain some editor unique identity, so that any hex can be related to the appropriate editor, perhaps automatically (e.g. MP signature at known location).This could be part of the version number?
+
+See “workflows” for discussion of how the user will be notified about the fact they are heading to an older version.
+
+(This kind of thing can enable the ability to show a user the exact version of the source code that was used to generate their hex file, including MicroPython code)
+
+## Release cadence
+
+Because we must commit to maintaining and fixing versions that are deployed on python.microbit.org it is desirable to make a small number of major release. 
+
+We propose an annual cadence for major releases, with an RC released early in the year so that a stable release can be made in July after the school year ends. This gives teachers time to become familiar with it before the school year start. The previous year’s version will remain available after this but will no longer be redirected to via python.microbit.org/latest or python.microbit.org/
```

Comment:
When I talked to some teachers about Mu, one of the things that was mentioned is that one of them liked the fact that the same version could be installed in all computers and stick with it for the rest of the year. In the same session, some of the teachers I was talking with were apprehensive to even consider PXT at all because they had already lessons planned with the original blocks editor. 
This is only a personal anecdote, but this limited feedback was from the teachers with the initiative to attend a local networking sessions on their own time, I imagine (and this is only a assumption from my part) that the "fear of change" might be more prevalent in the wider teaching community.

> I want to stress that the wider Python community won't follow these cadences

If the front page at `python.microbit.org` makes it clear that there are modern versions available at the `python.microbit.org/latest` or versioned URL I'd be surprised if we have this problem.
We cannot really count on toasts or modals/pop-ups to be read and action correctly by the average classroom user, but I think we can count on this to spread the message of "shiner things" to the people with a personal interest.


I do agree that the way this "release procedure" is described here kind of puts all the effort (which we should remember is coming from a community of volunteers) into a specific time of the year, and as much as we all love working on this project, life happens and there are other priorities.
If we continuously develop through the year, once we get closer to this July time-frame we can consciously decide if we can push a little harder towards a specific goal. If we have been releasing features in their own versioned URLs through the year, then they should be stable enough to make it to the main release, and we can then decide which version has been "hardened" enough to be released as the next academic year release.